### PR TITLE
feat: create ion tour

### PR DESCRIPTION
**Pull request by vinicius-guedes-brisa**
## Description

A popup component for guiding users through a product.

## Screenshots

https://github.com/user-attachments/assets/60f19e1f-ea89-4b1e-8c2f-a2f7d10d23c3

https://github.com/user-attachments/assets/027eb58b-9de8-4684-aa24-6c9e24a9cceb

## View Storybook

[Link](https://62eab350a45bdb0a5818520e-xfvabxkehk.chromatic.com/?path=/story/ion-data-display-tour--basic-tour)

## Compliance

- [x] I have verified that this change complies with our code and contribution policies.
- [x] I have verified that this change does not cause regressions and does not affect other parts of the code.

image

### CODE REVIEW

**Review by matheus-maximiano-brisa — APPROVED** [review]

File: projects/ion/src/lib/tour/tour.service.ts [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+  }
+
+  public removeStep(stepId: IonTourStepProps['ionStepId']): void {
+    this._tours[this.activeTourId].delete(stepId);
+  }
+
+  public start(props: IonStartTourProps = {}): void {
+    setTimeout(() => {
+      if (isEmpty(this._tours)) {
+        console.error('No steps found!');
```
Comment (danilo-moreira-brisa) [inline projects/ion/src/lib/tour/tour.service.ts:77]: Disable linter on this line if the console it's really intentional (same for consoles below).

File: stories/TourStep.stories.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import { CommonModule } from '@angular/common';
+import { Meta, Story } from '@storybook/angular';
+
+import { IonTourModule } from '../projects/ion/src/lib/tour';
+import { TourStepDemoComponent } from '../projects/ion/src/lib/tour/mocks/tour-step-props.component';
+import {
+  IonSharedModule,
+  PopoverPosition,
+  IonTourStepProps,
```
Comment (danilo-moreira-brisa) [inline stories/TourStep.stories.ts:9]:
```suggestion

```

**Review by danilo-moreira-brisa — APPROVED** [review]
Amazing work!

**Review by allan-chagas-brisa — APPROVED** [review]

File: projects/ion/src/lib/tour/tour-backdrop/tour-backdrop.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  props: Partial<IonTourBackdropComponent> = {}
+): Promise<RenderResult<IonTourBackdropComponent>> => {
+  return render(IonTourBackdropComponent, {
+    declarations: [IonTourBackdropComponent],
+    componentProperties: {
+      ...DEFAULT_PROPS,
+      ...props,
+    },
+  });
+};
```
Comment (pedro-martins-brisa) [inline projects/ion/src/lib/tour/tour-backdrop/tour-backdrop.component.spec.ts:18]: DEFAULT_PROPS is empty and is only used in the SUT.

```suggestion
const sut = async (
  props: Partial<IonTourBackdropComponent> = {}
): Promise<RenderResult<IonTourBackdropComponent>> => {
  return render(IonTourBackdropComponent, {
    declarations: [IonTourBackdropComponent],
    componentProperties: {
      ...props,
    },
  });
};
```

**Review by pedro-martins-brisa — APPROVED** [review]

**Pull request merged**